packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts

```diff
@@ -118,17 +85,38 @@ export class WalletLinkSigner implements Signer {
     }
   }
 
-  private async watchAsset(
-    type: string,
-    address: string,
-    symbol?: string,
-    decimals?: number,
-    image?: string,
-    chainId?: number
-  ): Promise<boolean> {
+  private async watchAsset(params: RequestParam) {
+    const request = (Array.isArray(params) ? params[0] : params) as {
+      type: string;
+      options: {
+        address: string;
+        symbol?: string;
+        decimals?: number;
+        image?: string;
+      };
+    };
+    if (!request.type) {
+      throw standardErrors.rpc.invalidParams('Type is required');
+    }
+
+    if (request?.type !== 'ERC20') {
+      throw standardErrors.rpc.invalidParams(`Asset of type '${request.type}' is not supported`);
+    }
+
+    if (!request?.options) {
+      throw standardErrors.rpc.invalidParams('Options are required');
+    }
+
+    if (!request?.options.address) {
+      throw standardErrors.rpc.invalidParams('Address is required');
+    }
+
+    const chainId = this.getChainId();
+    const { address, symbol, image, decimals } = request.options;
```

Comment:
consolidated to simplify call stack
<img width="819" alt="image" src="https://github.com/user-attachments/assets/3fb27231-9047-4282-b6ce-ac3cbff2ab58">